### PR TITLE
Add alerts on saved searches

### PR DESCRIPTION
**Pull request by trasher**
Factorize search result count; add count on alerts form
Add crontask and notification, its targets and its template

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: front/savedsearch.form.php [hunk 1]
```diff
@@ -60,6 +60,28 @@
    $savedsearch->check($_POST['id'], UPDATE);
    $savedsearch->update($_POST);
    Html::back();
+} else if (isset($_GET['create_notif'])) {
+   $savedsearch->check($_GET['id'], UPDATE);
+   $notif = new Notification();
+   $notif->getFromDBByCrit(['event' => 'alert_' . $savedsearch->getID()]);
+
+   if ($notif->isNewItem()) {
```
Comment (orthagh) [inline front/savedsearch.form.php]: maybe move this code into a method of SavedSearch ?

File: inc/savedsearch_alert.class.php [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+            $count = $data['data']['totalcount'];
+         }
+      } catch (\RuntimeException $e) {
+         $pass = true;
+      }
+
+      $this->showFormHeader($options);
+
+      if ($this->isNewID($ID)) {
+         echo "<input type='hidden' name='savedsearches_id' value='".$options['savedsearches_id']."'>";
```
Comment (orthagh) [inline inc/savedsearch_alert.class.php]: Use ```echo Html::hidden(...)```

File: install/mysql/glpi-9.2-empty.sql [hunk 3]
```diff
@@ -251,6 +251,27 @@ CREATE TABLE `glpi_savedsearches_users` (
...
 
+### Dump table glpi_savedsearches_alerts
+
+DROP TABLE IF EXISTS `glpi_savedsearches_alerts`;
+CREATE TABLE `glpi_savedsearches_alerts` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `savedsearches_id` int(11) NOT NULL DEFAULT '0',
+  `name` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
+  `is_active` tinyint(1) NOT NULL DEFAULT '0',
+  `operator` tinyint(1) NOT NULL,
```
Comment (orthagh) [inline install/mysql/glpi-9.2-empty.sql]: is there a gain to use an int for operator instead directly a varchar ? 
varchar could permit a more readable db, no?

Comment (trasher) [inline install/mysql/glpi-9.2-empty.sql]: A tinyint, not an int :p Well, this is lighter than a varchar as far as I remember.

Making DB readable is not really something I care about generally; as far as we use constants in code.

**Comment by trasher** [comment]
A bit of documentation on saved searches and their alerts has been submitted on the user documentation; see https://github.com/glpi-project/doc/pull/11

**Pull request merged**